citest/json_predicate/binary_predicate.py

```diff
@@ -146,6 +150,83 @@ def __call__(self, context, value):
                            path_value=PathValue('', value), valid=valid)
 
 
+class DictMatchesPredicate(BinaryPredicate):
+  """Implements binary predicate comparison predicates against dict values.
+
+  The dictionary value is in the form {field : pred} where each element
+  in the dictionary is a predicate for validating that particular field.
+  A strict predicate means that exactly all specified fields must be present.
+  Otherwise this permits additional fields with arbitrary values.
+  """
+
+  @property
+  def strict(self):
+    """Whether only the specified fields may be present (True) or not."""
+    return self.__strict
+
+  def __init__(self, operand, **kwargs):
+    """Constructor."""
+    if not isinstance(operand, dict):
```

Comment:
All the BinaryPredicates are constructed with an operand, which is the "operand" in the binary operator that you are comparing against. The other operand is passed into the __call__ method as the value when it comes time to test a particular value.